modules/codegen-plugin/src/smithy4s/codegen/Smithy4sCodegenPlugin.scala

```diff
@@ -131,13 +142,18 @@ object Smithy4sCodegenPlugin extends AutoPlugin {
     val excludedNamespaces =
       (conf / smithy4sExcludedNamespaces).?.value.map(_.toSet)
     val updateReport = (conf / update).value
-    val internalDependencyJars =
-      (conf / internalDependencyAsJars).value.seq
-        .map(_.data)
-        .map(os.Path(_))
-        .toList
+
+    val localDependencyJars = Def.taskIf {
+      if ((conf / smithy4sAggregateLocalDependencies).value) {
```

Comment:
I wonder if rather than having a flag setting, we should make a TaskKey for this and give users the control over the task itself. This would increase consistency between SBT and Mill and would give more control to the user, if they wanted to add some unrelated jars, or have a more fine-grain filtering than "all or nothing"

Reply:
yeah, that makes is easier on our side too. I suppose it's better this way because the non-default option is just `Nil`, not another task or something.